### PR TITLE
Auto Accept Room Invites

### PR DESCRIPTION
**Pull request by ismailgulek**
Introduces `autoAcceptRoomInvites` flag on `MXSDKOptions` which is false by default.

### CODE REVIEW

File: MatrixSDK/Background/MXBackgroundSyncService.swift [hunk 1]
```diff
@@ -504,6 +504,21 @@ public enum MXBackgroundSyncServiceError: Error {
...
+        if MXSDKOptions.sharedInstance().autoAcceptRoomInvites,
+           let invitedRooms = syncResponse.rooms?.invite {
+            invitedRooms.forEach { roomId, roomSync in
+                MXLog.debug("[MXBackgroundSyncService] handleSyncResponse: Auto-accepting room invite for \(roomId)")
+                restClient.joinRoom(roomId) { response in
+                    switch response {
+                    case .success:
+                        MXLog.debug("[MXBackgroundSyncService] handleSyncResponse: Joined room: \(roomId)")
+                    case .failure(let error):
+                        MXLog.error("[MXBackgroundSyncService] handleSyncResponse: Failed to join room: \(roomId), error: \(error)")
```
Comment (manuroe) [inline MatrixSDK/Background/MXBackgroundSyncService.swift:516]: Do you know if the `/join` request generates an error if we double join a room?
This is what will happen if the auto-join is firstly managed by the NSE.

**Review by manuroe — COMMENTED** [review]
With https://github.com/vector-im/element-ios/pull/4452, invites will be hidden forever if there is a network failure on `/join`.

We need another mechanism to retry. We can do it on start but it will be great to be able to retry while the app is running.

File: MatrixSDK/Background/MXBackgroundSyncService.swift (hunk 1 above)
Comment (ismailgulek) [reply MatrixSDK/Background/MXBackgroundSyncService.swift:516]: You're right. I've checked this and seems we don't have an error. I need to fix this.

Comment (ismailgulek) [reply MatrixSDK/Background/MXBackgroundSyncService.swift:516]: Error added with https://github.com/matrix-org/matrix-ios-sdk/pull/1134/commits/260d68416d5d0c92914b48af08c2182468a8f898

Comment (ismailgulek) [reply MatrixSDK/Background/MXBackgroundSyncService.swift:516]: Retry mechanism added with https://github.com/matrix-org/matrix-ios-sdk/pull/1134/commits/93aca3b8481d8d606dee8bb99c2984e8a4182feb

File: MatrixSDK/MXSession.m [hunk 2]
```diff
@@ -2217,6 +2224,21 @@ - (MXHTTPOperation*)joinRoom:(NSString*)roomIdOrAlias
                      success:(void (^)(MXRoom *room))success
                      failure:(void (^)(NSError *error))failure {
     
+    if ([self isJoinedOnRoom:roomIdOrAlias])
```
Comment (manuroe) [inline MatrixSDK/MXSession.m:2227]: Why not using `RoomSummary.membershipTransitionState` and `MXMembershipTransitionStateJoining`?

Comment (ismailgulek) [reply MatrixSDK/MXSession.m:2227]: Updated for both is joined and is invited logic.

**Review by manuroe — APPROVED** [review]

**Pull request merged**